source/handshake.c

```diff
@@ -139,9 +142,11 @@ static void do_handshake_ossl_lib_ctx_per_thread(size_t num)
                                              SSL_ERROR_NONE);
         perflib_shutdown_ssl_connection(serverssl, clientssl);
         serverssl = clientssl = NULL;
-        SSL_CTX_free(lsctx);
-        SSL_CTX_free(lcctx);
-        lsctx = lcctx = NULL;
+        if (share_ctx == 0) {
+            SSL_CTX_free(lsctx);
+            SSL_CTX_free(lcctx);
+            lsctx = lcctx = NULL;
+        }
```

Comment:
This is a nit, given the code here is just for performance testing: once we exit the loop we should also release lsctx/lcctx we allocate in the first iteration. otherwise looks good.